Snowflake.Data/Core/FileTransfer/StorageClient/SFRemoteStorageUtil.cs

```diff
@@ -464,6 +466,11 @@ internal static async Task DownloadOneFileAsync(SFFileMetadata fileMetadata, Can
                     fileMetadata.destFileSize = fileInfo.Length;
                     return;
                 }
+                else if (fileMetadata.resultStatus == ResultStatus.RENEW_TOKEN.ToString() ||
```

Comment:
Reverting the change that moved the if clause into a function: https://github.com/snowflakedb/snowflake-connector-net/commit/d061cb65e8d4a3d601db99a20854c3fc6359b761#diff-141fae8d9ec0879939309639fad147fd50b1bba1902c0dbd13fd25fb5c10e3c7L501

Originally, return is called when the result status was `RENEW_TOKEN` or `RENEW_PRESIGNED_URL` and the loop stops, the token would get renewed and then try again

However, with the change, the return statement is called within the `HandleDownloadFileErr` function so the loop continues with the expired token instead of stopping